File: HealthSysHub.Web.UI/Services/ConsultationService.cs
using HealthSysHub.Web.UI.Factory;
using HealthSysHub.Web.UI.Interfaces;
using HealthSysHub.Web.UI.Models;
using HealthSysHub.Web.Utility.Models;

namespace HealthSysHub.Web.UI.Services
{
    public class ConsultationService : IConsultationService
    {
        private readonly IRepositoryFactory _repository;

        public ConsultationService(IRepositoryFactory repository)
        {
            _repository = repository;
        }

        public async Task<List<ConsultationDetails>> GetConsultationDetailsAsync()
        {
            return await _repository.SendAsync<List<ConsultationDetails>>(HttpMethod.Get, "Consultation/GetAllConsultationsAsync");
        }

        public async Task<List<ConsultationDetails>> GetConsultationDetailsByAppointmentIdAsync(Guid appointmentId)
        {
            var uri = Path.Combine("Consultation/GetConsultationByAppointmentIdAsync", appointmentId.ToString());
            return await _repository.SendAsync<List<ConsultationDetails>>(HttpMethod.Get, uri);
        }

        public async Task<List<ConsultationDetails>> GetConsultationDetailsByConsultationIdAsync(Guid consultationId)
        {
            var uri = Path.Combine("Consultation/GetConsultationByConsultationIdAsync", consultationId.ToString());
            return await _repository.SendAsync<List<ConsultationDetails>>(HttpMethod.Get, uri);
        }

        public async Task<List<ConsultationDetails>> GetConsultationDetailsByDoctorAsync(Guid doctorId)
        {
            var uri = Path.Combine("Consultation/GetConsultationsByDoctorAsync", doctorId.ToString());
            return await _repository.SendAsync<List<ConsultationDetails>>(HttpMethod.Get, uri);
        }

        public async Task<List<ConsultationDetails>> GetConsultationDetailsByHospitalAsync(Guid hospitalId)
        {
            var uri = Path.Combine("Consultation/GetConsultationsByHospitalAsync", hospitalId.ToString());
            return await _repository.SendAsync<List<ConsultationDetails>>(HttpMethod.Get, uri);
        }

        public async Task<List<Consultation>> GetConsultationsAsync()
        {
            return await _repository.SendAsync<List<Consultation>>(HttpMethod.Get, "Consultation/GetConsultationsAsync");
        }

        public async Task<List<Consultation>> GetConsultationsByDoctorAsync(Guid doctorId)
        {
            var uri = Path.Combine("Consultation/GetConsultationsByDoctorAsync", doctorId.ToString());
            return await _repository.SendAsync<List<Consultation>>(HttpMethod.Get, uri);
        }

        public async Task<List<Consultation>> GetConsultationsByHospitalAsync(Guid hospitalId)
        {
            var uri = Path.Combine("Consultation/GetConsultationsByHospitalAsync", hospitalId.ToString());
            return await _repository.SendAsync<List<Consultation>>(HttpMethod.Get, uri);
        }

        public async Task<Consultation> InsertOrUpdateConsultationAsync(Consultation consultation)
        {
            return await _repository.SendAsync<Consultation, Consultation>(HttpMethod.Post, "Consultation/InsertOrUpdateConsultationAsync", consultation);
        }

        public async Task<ConsultationDetails> InsertOrUpdateConsultationDetailsAsync(ConsultationDetails consultationDetails)
        {
            return await _repository.SendAsync<ConsultationDetails, ConsultationDetails>(HttpMethod.Post, "Consultation/InsertOrUpdateConsultationDetailsAsync", consultationDetails);
        }
    }
}

File: healthsyshub.web.ip.ui/src/global/API/consultation.service.js
import { send } from '@/global/API/api.service'
import { endpoints } from '@/environment'

export default {
    // Get all consultations
    async GetAllConsultationsAsync() {
        const url = endpoints.UrlConstants.GetAllConsultationsAsync;
        const response = await send("GET", url);
        return response;
    },

    // Get consultation by appointment ID
    async GetConsultationByAppointmentIdAsync(appointmentId) {
        const url = `${endpoints.UrlConstants.GetConsultationByAppointmentIdAsync}/${appointmentId}`;
        const response = await send("GET", url);
        return response;
    },

    // Get consultation by consultation ID
    async GetConsultationByConsultationIdAsync(consultationId) {
        const url = `${endpoints.UrlConstants.GetConsultationByConsultationIdAsync}/${consultationId}`;
        const response = await send("GET", url);
        return response;
    },

    // Get consultations by doctor ID
    async GetConsultationsByDoctorAsync(doctorId) {
        const url = `${endpoints.UrlConstants.GetConsultationsByDoctorAsync}/${doctorId}`;
        const response = await send("GET", url);
        return response;
    },

    // Get consultations by hospital ID
    async GetConsultationsByHospitalAsync(hospitalId) {
        const url = `${endpoints.UrlConstants.GetConsultationsByHospitalAsync}/${hospitalId}`;
        const response = await send("GET", url);
        return response;
    },

    // Create or update consultation details
    async InsertOrUpdateConsultationDetailsAsync(consultationDetails) {
        const url = endpoints.UrlConstants.InsertOrUpdateConsultationDetailsAsync;
        const response = await send("POST", url, consultationDetails);
        return response;
    }
}